x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/transforms/DataFrameIndexer.java

```diff
@@ -69,6 +72,10 @@ protected void onStart(long now, ActionListener<Void> listener) {
                 agg.getBuckets().isEmpty());
     }
 
+    protected boolean isFirstRun() {
+        return getPosition() == null;
```

Comment:
This will obviously have to change as we start allowing re-runs of the indexer on new checkpoints